library/src/main/java/com/otaliastudios/zoom/OverZoomRangeProvider.kt

```diff
@@ -0,0 +1,27 @@
+package com.otaliastudios.zoom
+
+/**
+ * Defines the allowed range for overzoom.
+ */
+interface OverZoomRangeProvider {
+
+    /**
+     * Calculates the maximum overzoom to allow
+     *
+     * @param engine the zoom engine
+     * @param inwards true for inwards, false for outwards
+     * @return the maximum overzoom to allow
+     */
+    @ZoomApi.RealZoom
+    fun getOverZoom(engine: ZoomEngine, inwards: Boolean): Float
+
+    companion object {
+        private const val DEFAULT_OVERZOOM_FACTOR = 0.1f
+        val DEFAULT = object : OverZoomRangeProvider {
```

Comment:
I wonder if we need some Jvm annotation to make this accessible from java as a static field?

Reply:
Hmm, yes indeed it does not work by default.
I tried to use `@JvmField`

but it states:
```
JvmField could be applied only if all interface companion properties are 'public final val' with '@JvmField' annotation
```
which is kinda odd imo. To solve this I moved the private constant into the definition of the default `OverZoomRangeProvider`.